### PR TITLE
Added vector columns

### PR DESCRIPTION
**Pull request by richardgill**
Adds vector column type.

The UI/UX is suboptimal as we discussed on Tuesday.

https://user-images.githubusercontent.com/6640874/227447312-ab658506-d636-45bd-83f9-381c37eb3765.mov

### CODE REVIEW

File: cli/src/commands/schema/edit.ts [hunk 1]
```diff
@@ -453,6 +457,28 @@ Beware that this can lead to ${chalk.bold(
...
+            state: ColumnEditState,
+            item: { value: string | undefined },
+            index: number
+          ) {
+            if (!isColumnAdded && value !== state.initial.vectorDimension) {
+              return `Cannot change the vector dimension of existing vector columns`;
+            }
+            if (state.values.type === 'vector') {
+              if (!value) {
+                return 'The vectorDimension field must be filled the columns of type `vector`';
```
Comment (tsg) [inline cli/src/commands/schema/edit.ts:474]: Is a word missing here?

```suggestion
                return 'The vectorDimension field must be filled for the columns of type `vector`';
```

**Review by tsg — COMMENTED** [review]
From the video it looks good, but somehow I couldn't test it with the `npx @xata.io/cli@0.0.0-alpha.va75ac26 schema edit` command, the vector type is not an option there. I assume I'm doing something wrong, but thought it's worth mentioning it.

Left a tiny comment on the wording, otherwise LGTM.

**Review by SferaDev — APPROVED** [review]
@tsg Weird, I tried it with `npx @xata.io/cli@0.0.0-alpha.va75ac26` and worked perfectly

**Comment by richardgill** [comment]
@tsg thanks for spotting the weird wording in the error. 

It was broken for link and I'd copied it. I've fixed it both places like this 👍 
https://github.com/xataio/client-ts/pull/962/commits/0a43af36bacda94e42cd0986cd40dc512ea9d392

**Review by tsg — APPROVED** [review]

**Pull request merged**